PancakeSwap-Bots/Large-swap/src/utils.ts

```diff
@@ -0,0 +1,90 @@
+import { ethers, Finding, FindingType, FindingSeverity } from "forta-agent";
+import { createAddress } from "forta-agent-tools/lib/tests.utils";
+import BigNumber from "bignumber.js";
+BigNumber.set({ DECIMAL_PLACES: 18 });
+
+const SWAP_EVENT =
+  "event Swap(address indexed sender, uint amount0In, uint amount1In, uint amount0Out,uint amount1Out,address indexed to)";
+
+const LARGE_THRESHOLD = "2.5"; // percent
+const ERC20ABI = ["function balanceOf(address account) public view returns (uint256)"];
+const PANCAKE_PAIR_ABI = [
+  "function token0() external view returns (address)",
+  "function token1() external view returns (address)",
+];
+const PANCAKE_FACTORY_ABI = ["function getPair(address tokenA, address tokenB) external view returns (address pair)"];
+const PANCAKE_FACTORY_ADDRESS = "0xcA143Ce32Fe78f1f7019d7d551a6402fC5350c73";
+
+const toBn = (ethersBn: ethers.BigNumberish) => new BigNumber(ethersBn.toString());
+
+const createContract = (contractAddress: string, abi: string | string[], provider: ethers.providers.Provider) =>
+  new ethers.Contract(contractAddress, abi, provider);
+
+const isValidPancakePair = async (
+  pairAddress: string,
+  provider: ethers.providers.Provider,
+  block: number,
+  pancakeFactoryAddr: string
+): Promise<[boolean, string, string]> => {
+  const pairContract = createContract(pairAddress, PANCAKE_PAIR_ABI, provider);
+  const token0Address: string = await pairContract.token0({ blockTag: block });
+  const token1Address: string = await pairContract.token1({ blockTag: block });
+  const pancakeFactory = createContract(pancakeFactoryAddr, PANCAKE_FACTORY_ABI, provider);
+  const tokenPair: string = await pancakeFactory.getPair(token0Address, token1Address, { blockTag: block });
+  const isValid =
+    tokenPair !== createAddress("0x0") && tokenPair.toLowerCase() === pairAddress.toLowerCase() ? true : false;
+  return [isValid, token0Address, token1Address];
```

Comment:
this is not the best way of validating pairs, please use the `create2` approach.